packages/core/src/client.test.ts

```diff
@@ -1265,6 +1265,22 @@ describe('Client', () => {
     expect(client.getCachedReference(createReference(result.entry?.[0]?.resource as Patient))).toBeDefined();
   });
 
+  test('Search and return 404', async () => {
+    console.error = jest.fn();
+    const fetch = mockFetch(404, {
+      resourceType: 'Bundle',
+      entry: [{ resource: { resourceType: 'Patient', id: '123' } }],
+    });
+    const client = new MedplumClient({ fetch });
+    try {
+      await client.search('Patient');
```

Comment:
More precise to check that the `OperationOutcomeError` is thrown:
https://jestjs.io/docs/tutorial-async